TechnologyAssessmentApp/TechnologyAssessmentApp/Services/AppApiManager.swift

```diff
@@ -10,22 +10,18 @@ import UIKit
 import Foundation
 
 class AppApiManager: NSObject {
-
-    func getArticlesList(completionHandler: @escaping (_ articleList: [Article]?, _ error: AppError?) -> Void) {
+    func getArticlesList(completionHandler: @escaping (_ articleList: ArticleViewModel?, _ error: AppError?)
+        -> Void) {
         let networkClient = AppNetworkClient.sharedInstance
         let url = WebUrl.baseUrl.replacingOccurrences(of: "{section}", with: "all-sections")
             .replacingOccurrences(of: "{period}", with: "7")
             .replacingOccurrences(of: "{sampleKey}", with: sampleKey)
-        networkClient.getDataFor(baseUrl: url, responseType: .json, endPoint: "", params: nil) { (json, customError) in
-            if customError == nil, let response = json as? [String: Any], response["status"] as? String == "OK" {
-                if let networkArticles = response["results"] as? [[String: Any]] {
-                    var articles: [Article] = []
-                    // Create article objects out of a JSON object
-                    for networkArticle in networkArticles {
-                        let article = Article(jsonObject: networkArticle)
-                        articles.append(article)
-                    }
-                  completionHandler(articles, nil)
+        networkClient.getDataFor(baseUrl: url,
+                                 responseType: .json, endPoint: "", params: nil) {(jsonData, customError) in
+            if customError == nil, let response = jsonData as? Data {
+                let articleModel = try? JSONDecoder().decode(ArticleViewModel.self, from: response)
+                if let article = articleModel, article.status == "OK" {
```

Comment:
`article.status == "OK"` any better way we could handle errors ? I didn't look at NY Times API, but maybe they have a list of error code ? or error string listed and we can create an enum for it ?